common-money.lic

```diff
@@ -40,26 +40,45 @@ module DRCM
     check_wealth(hometown_currency(hometown))
   end
 
-  def ensure_copper_on_hand(copper, hometown)
-    on_hand = wealth(hometown)
+  def ensure_copper_on_hand(copper, settings)
+    on_hand = wealth(settings.hometown)
     return true if on_hand >= copper
     withdrawals = minimize_coins(copper - on_hand)
-    withdrawals.all? { |amount| withdraw_exact_amount?(amount, hometown) }
+    withdrawals.all? { |amount| withdraw_exact_amount?(amount, settings) }
   end
 
-  def withdraw_exact_amount?(amount_as_string, hometown)
-    bank_id = get_data('town')[hometown]['deposit']['id']
+  def withdraw_exact_amount?(amount_as_string, settings)
+    if settings.bankbot_enabled
+      bank_id = settings.bankbot_room_id
+    else  
+      bank_id = get_data('town')[settings.hometown]['deposit']['id']
+    end  
     DRCT.walk_to(bank_id)
     DRC.release_invisibility
     loop do
-      case DRC.bput("withdraw #{amount_as_string}", 'The clerk counts', 'The clerk tells', 'The clerk glares at you.', 'You count out', 'find a new deposit jar', 'If you value your hands', 'Hey!  Slow down!', "You must be at a bank teller's window to withdraw money")
-      when 'The clerk counts', 'You count out'
-        break true
-      when 'The clerk glares at you.', 'Hey!  Slow down!'
-        pause 15
-      when 'The clerk tells', 'If you value your hands', 'find a new deposit jar', "You must be at a bank teller's window to withdraw money"
-        break false
-      end
+      if settings.bankbot_enabled && DRRoom.pcs.include?(settings.bankbot_name)
+        amount_convert, type = amount_as_string.split(" ")
+        amount = convert_to_copper(amount_convert, type)
+        currency = hometown_currency(settings.hometown)
+        case DRC.bput("whisper #{settings.bankbot_name} withdraw #{amount} #{currency}", "offers you", "Whisper what to who?")
+        when "offers you"
+          DRC.bput("accept tip", "You accept")
+          break true
+        when "Whisper what to who?"
+          break false
+        end     
+      else
```

Comment:
It feels like this would be cleaner with a function that handles the bankbot case and returns true/false. Then you call the "real" bank case if the bankbot function returned false. What do you think?

Reply:
I changed it once to flow better but it broke everything. I can look into again.

Reply:
Did you try this change?

Reply:
I did on my bankbot guys but on Dartellum and Ceindrech it hung them. I know why now, I had no logic for the else for the first if statement. I will make a function and call it in both sections.